olive/model/__init__.py

```diff
@@ -588,7 +588,7 @@ def prepare_session(
         execution_providers: Union[str, List[str]] = None,
         rank: Optional[int] = None,
     ):
-        return self.load_model().eval()
+        return self.load_model(device).eval()
```

Comment:
I think it might be better to do the `to("cuda")` directly here instead of passing it to `load_model`. In this code, the `device` is only used when using hf_config to load the model. 
Also think it's better to not add `device` argument to the hf model loader func.

Reply:
The reason I didn't add "to("cuda")" here is I'm not sure how other `load_model` works and if they support `to("cuda")`, especially loading from module_loader.

Reply:
btw, why is this change need in this PR? We do the `to("cuda")` step for the evaluator here https://github.com/microsoft/Olive/blob/main/olive/evaluator/olive_evaluator.py#L687 

I think we can assume that the loaded data is a torch model so they should have `to("cuda")` support.

Reply:
Maybe, we can do a try/except with the `to("cuda")` in the prepare session to be safe.

Reply:
Yeah you are right I just double confirmed huggingface supports both https://github.com/huggingface/optimum/issues/336. I'll remove this device switch logic since we can do this in evaluator.

Reply:
Thanks for checking!